pkg/apis/monitoring/v1/types.go

```diff
@@ -426,6 +432,24 @@ type PrometheusRuleExcludeConfig struct {
 	RuleName string `json:"ruleName"`
 }
 
+// PrometheusObjectReference prometheus object reference
+type PrometheusObjectReference struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion,omitempty"`
+	// Kind of the referent.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `json:"kind,omitempty"`
+	// Namespace of the referent.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
+	Namespace string `json:"namespace,omitempty"`
+	// LabelSelector to select prometheus objects.
+	LabelSelector metav1.LabelSelector `json:"selector"`
```

Comment:
Can you expand on the need for using label selector here? I would expect that there are few objects that should be excluded from enforcement.
Also I think that we should keep the ability to select objects by namespace/name.

Reply:
Sure - the use-case here is when the prometheus-operator is run in a multi-tenant setup where some of the prometheus exporters scraped by Prometheus are deployed in a trusted namespace and scrape metrics from all other tenant namespaces. A good example is kube-state-metrics.

We end up by having the following Prometheus spec:

```
prometheusRulesExcludedFromEnforce:
    - ruleName: prometheus-promoperator-alertmanager.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-general.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-k8s.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-apiserver.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-apiserver-availability.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-apiserver-error
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-apiserver-slos
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-prometheus-general.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-prometheus-node-alerting.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-prometheus-node-recording.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-scheduler.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kube-state-metrics
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubelet.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-absent
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-resources
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-storage
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-system
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-system-apiserver
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-system-kubelet
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-system-controller-manager
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-system-scheduler
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-node-exporter.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-node-exporter
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-node.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-node-network
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-node-time
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-prometheus-operator
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-prometheus.rules
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-prometheus
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-kubernetes-apps
      ruleNamespace: ns-team-aep-pipeline-monitoring
    - ruleName: prometheus-promoperator-etcd
      ruleNamespace: ns-team-aep-pipeline-monitoring
```

While we do generate this automatically it would be really handy if we could label and select these "trusted" rules via labels